### PR TITLE
Mui fine foods products

### PR DESCRIPTION
**Pull request by ozkalai**
Test me! 'MUI'
[Link to MUI-FINE-FOOD-PRODUCT](https://mui-fin-refine.pankod.com)

Hey, I'm client! Test me! 'MUI'
[Link to MUI-FINE-FOOD-PRODUCT](https://mui-fin-client.refine.pankod.com)

Test me! 'MUI'
[Link to MUI-FINE-FOOD-PRODUCT](https://mui-fin-refine.pankod.com)

Hey, I'm client! Test me! 'MUI'
[Link to MUI-FINE-FOOD-PRODUCT](https://mui-fin-client.refine.pankod.com)

Test me! 'MUI'
[Link to MUI-FINE-FOOD-PRODUCT](https://mui-fin-refine.pankod.com)

Hey, I'm client! Test me! 'MUI'
[Link to MUI-FINE-FOOD-PRODUCT](https://mui-fin-client.refine.pankod.com)

Test me! 'MUI'
[Link to MUI-FINE-FOOD-PRODUCT](https://mui-fin-refine.pankod.com)

Hey, I'm client! Test me! 'MUI'
[Link to MUI-FINE-FOOD-PRODUCT](https://mui-fin-client.refine.pankod.com)

Test me! 'MUI'
[Link to MUI-FINE-FOOD-PRODUCT](https://mui-fin-refine.pankod.com)

Test me! 'MASTER'
[Link to MUI-FINE-FOOD-PRODUCT](https://mui-fin-refine.pankod.com)

**IMPORTANT: Please do not create a Pull Request without creating an issue first.**

*Any change needs to be discussed before proceeding. Failure to do so may result in the rejection of the pull request.*

Please provide enough information so that others can review your pull request:

Explain the **details** for making this change. What existing problem does the pull request solve?

**Test plan (required)**

Demonstrate the code is solid. If not, please add `WIP:` in its title.

**Closing issues**

Put `closes #XXXX` in your comment to auto-close the issue that your PR fixes (if such).

### CODE REVIEW

File: examples/fineFoods/admin/mui/src/pages/products/list.tsx [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+                                price={i.price}
+                                id={i.id}
+                                description={i.description}
+                                isActive={i.isActive}
+                                createdAt={i.createdAt}
+                                category={i.category}
+                                stock={i.stock}
+                            />
+                        </Grid>
+                    ))}
```
Comment (omeraplak) [inline examples/fineFoods/admin/mui/src/pages/products/list.tsx:44]:
```suggestion
                {tableQueryResult.data?.data.map((i: IProduct) => (
                        <Grid item xs={2} sm={3} md={3} key={i.id}>
                            <ProductItem
                                name={i.name}
                                key={i.id}
                                images={i.images}
                                price={i.price}
                                id={i.id}
                                description={i.description}
                                isActive={i.isActive}
                                createdAt={i.createdAt}
                                category={i.category}
                                stock={i.stock}
                            />
                        </Grid>
                    ))}
```

Comment (omeraplak) [reply examples/fineFoods/admin/mui/src/pages/products/list.tsx:44]: would it be better?

File: examples/fineFoods/admin/mui/src/components/product/categoryFilter.tsx [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+    categoryData,
+    isLoading,
+}) => {
+    const t = useTranslate();
+
+    const [filterCategories, setFilterCategories] = useState<string[]>([]);
+
+    useEffect(() => {
+        setFilters?.([
+            { ...filters[0] },
```
Comment (omeraplak) [inline examples/fineFoods/admin/mui/src/components/product/categoryFilter.tsx:27]:
```suggestion
            { ...filters[0] },
```
why are we doing this?

File: examples/fineFoods/admin/mui/src/components/product/create.tsx [hunk 3]
```diff
@@ -0,0 +1,281 @@
...
+    visible,
+    close,
+    register,
+    handleSubmit,
+    categoryData,
+    setValue,
+}) => {
+    const t = useTranslate();
+
+    const [uploadedImage, setUploadedImage] = useState(
```
Comment (omeraplak) [inline examples/fineFoods/admin/mui/src/components/product/create.tsx:62]: Do we really need state?

File: examples/fineFoods/admin/mui/src/components/product/create.tsx [hunk 4]
```diff
@@ -0,0 +1,281 @@
...
+            }}
+            open={visible}
+            onClose={() => close()}
+            anchor="right"
+        >
+            <Stack marginX="30px" marginY="20px">
+                <IconButton onClick={() => close()} sx={{ width: "30px" }}>
+                    <CloseIcon />
+                </IconButton>
+                <Typography fontWeight="bold">{t("Create Product")}</Typography>
```
Comment (omeraplak) [inline examples/fineFoods/admin/mui/src/components/product/create.tsx:134]: do we have such a translate key? By the way, why don't we use the "create" crud component?

File: examples/fineFoods/admin/mui/src/pages/products/list.tsx [hunk 5]
```diff
@@ -0,0 +1,209 @@
...
+        useTable<IProduct>({
+            resource: "products",
+            initialPageSize: 12,
+            initialFilter: [
+                { field: "name", operator: "contains", value: "" },
+                { field: "category.id", operator: "contains", value: "" },
+            ],
+        });
+
+    const paginationCount =
```
Comment (omeraplak) [inline examples/fineFoods/admin/mui/src/pages/products/list.tsx:54]: Can useTable provide the paginationCount? @salihozdemir

File: examples/fineFoods/admin/mui/src/pages/products/list.tsx [hunk 6]
```diff
@@ -0,0 +1,209 @@
...
+        tableQueryResult.data && Math.ceil(tableQueryResult.data?.total / 12);
+
+    return (
+        <>
+            {categories?.data && (
```
Comment (omeraplak) [inline examples/fineFoods/admin/mui/src/pages/products/list.tsx:59]: What exactly does it have to do with categories?

File: examples/fineFoods/admin/mui/src/pages/products/list.tsx [hunk 7]
```diff
@@ -0,0 +1,209 @@
...
+                                    onChange={(
+                                        e: React.ChangeEvent<HTMLInputElement>,
+                                    ) => {
+                                        setFilters([
+                                            {
+                                                field: "name",
+                                                operator: "contains",
+                                                value: e.target.value,
+                                            },
+                                            { ...filters[1] },
```
Comment (omeraplak) [inline examples/fineFoods/admin/mui/src/pages/products/list.tsx:125]: why are we doing this?

File: examples/fineFoods/admin/mui/src/pages/products/list.tsx [hunk 8]
```diff
@@ -0,0 +1,209 @@
...
+                                setCurrent(page);
+                            }}
+                        />
+                    </Grid>
+                    <Grid item md={4}>
+                        <Stack padding="8px">
+                            <Typography sx={{ fontWeight: 500 }}>
+                                {t("stores.tagFilterDescription")}
+                            </Typography>
+                            {categories?.data && (
```
Comment (omeraplak) [inline examples/fineFoods/admin/mui/src/pages/products/list.tsx:195]: why are we doing this?

File: examples/fineFoods/admin/mui/src/App.tsx [hunk 9]
```diff
@@ -19,12 +19,14 @@ import {
     StarBorderOutlined,
 } from "@mui/icons-material";
 import PeopleOutlineIcon from "@mui/icons-material/PeopleOutline";
+import LocalPizzaIcon from "@mui/icons-material/LocalPizza";
```
Comment (biskuvit) [inline examples/fineFoods/admin/mui/src/App.tsx:22]:
```suggestion
import PeopleOutline, LocalPizza from "@mui/icons-material";
```

File: examples/fineFoods/admin/mui/src/components/product/create.tsx [hunk 10]
```diff
@@ -0,0 +1,366 @@
...
+    OutlinedInput,
+    InputAdornment,
+    FormHelperText,
+    Create,
+    useAutocomplete,
+    SaveButtonProps,
+    TextField,
+} from "@pankod/refine-mui";
+
+import CloseIcon from "@mui/icons-material/Close";
```
Comment (biskuvit) [inline examples/fineFoods/admin/mui/src/components/product/create.tsx:40]:
```suggestion
import Close from "@mui/icons-material";
```

**Review by biskuvit — APPROVED** [review]

File: examples/fineFoods/admin/mui/src/components/product/create.tsx [hunk 11]
```diff
@@ -0,0 +1,366 @@
...
+                                            renderInput={(params) => (
+                                                <TextField
+                                                    {...params}
+                                                    label="Category"
+                                                    margin="normal"
+                                                    variant="outlined"
+                                                    error={!!errors.item}
+                                                    helperText={
+                                                        errors.item &&
+                                                        "Categoty required"
```
Comment (salihozdemir) [inline examples/fineFoods/admin/mui/src/components/product/create.tsx:314]:
```suggestion
                                                        errors.category.message
```

File: examples/fineFoods/admin/mui/src/components/product/edit.tsx [hunk 12]
```diff
@@ -0,0 +1,364 @@
...
+                                            renderInput={(params) => (
+                                                <TextField
+                                                    {...params}
+                                                    label="Category"
+                                                    margin="normal"
+                                                    variant="outlined"
+                                                    error={!!errors.item}
+                                                    helperText={
+                                                        errors.item &&
+                                                        "Categoty required"
```
Comment (salihozdemir) [inline examples/fineFoods/admin/mui/src/components/product/edit.tsx:312]:
```suggestion
                                                        errors.category.message
```

File: examples/fineFoods/admin/mui/src/pages/products/list.tsx [hunk 13]
```diff
@@ -0,0 +1,220 @@
+import React from "react";
+
```
Comment (salihozdemir) [inline examples/fineFoods/admin/mui/src/pages/products/list.tsx:2]:
```suggestion
```

File: examples/fineFoods/admin/mui/src/pages/products/list.tsx [hunk 14]
```diff
@@ -0,0 +1,220 @@
...
+import {
+    useTranslate,
+    IResourceComponentsProps,
+    useTable,
+} from "@pankod/refine-core";
+
```
Comment (salihozdemir) [inline examples/fineFoods/admin/mui/src/pages/products/list.tsx:8]:
```suggestion
```

File: examples/fineFoods/admin/mui/src/pages/products/list.tsx [hunk 15]
```diff
@@ -0,0 +1,220 @@
...
+import { useModalForm } from "@pankod/refine-react-hook-form";
+
```
Comment (salihozdemir) [inline examples/fineFoods/admin/mui/src/pages/products/list.tsx:10]:
```suggestion
```

File: examples/fineFoods/admin/mui/src/pages/products/list.tsx [hunk 16]
```diff
@@ -0,0 +1,220 @@
...
+    Paper,
+    Typography,
+    Box,
+    InputBase,
+    IconButton,
+    Stack,
+    Pagination,
+    Button,
+} from "@pankod/refine-mui";
+
```
Comment (salihozdemir) [inline examples/fineFoods/admin/mui/src/pages/products/list.tsx:22]:
```suggestion
```

**Review by salihozdemir — CHANGES_REQUESTED** [review]

File: examples/fineFoods/admin/mui/src/components/product/create.tsx [hunk 17]
```diff
@@ -66,8 +61,8 @@ export const CreateProduct: React.FC<CreateProductProps> = ({
     onFinish,
     saveButtonProps,
     control,
-    reset,
 }) => {
+    console.log("errors", errors);
```
Comment (salihozdemir) [inline examples/fineFoods/admin/mui/src/components/product/create.tsx:65]: console

**Review by omeraplak — APPROVED** [review]

**Review by salihozdemir — APPROVED** [review]

File: examples/fineFoods/admin/mui/src/components/product/create.tsx [hunk 18]
```diff
@@ -0,0 +1,365 @@
...
+    register,
+    handleSubmit,
+    setValue,
+    errors,
+    watch,
+    onFinish,
+    saveButtonProps,
+    control,
+}) => {
+    console.log("errors", errors);
```
Comment (omeraplak) [inline examples/fineFoods/admin/mui/src/components/product/create.tsx:65]:
```suggestion
```

**Pull request merged**